336 profile page

zlrkw11 opened the pull request:
closes #336

zlrkw11 commented:
![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/2f3829d0-bc61-4844-b8fc-13b611594dc9)
- default desktop view

zlrkw11 commented:
small issue, button hover effect only applied on the text, not bg color.

zlrkw11 commented:
![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/027484d4-0d54-4a5b-bddc-33ff44e8c804)
- mobile version

zlrkw11 commented:
![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/0c0d9cf2-c8f4-4aca-9dd1-14048cd176c1)
- sign out button takes user back to the login page

bcho892 commented:
> ![image](https://private-user-images.githubusercontent.com/64724157/331859891-027484d4-0d54-4a5b-bddc-33ff44e8c804.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxMTAzNTUsIm5iZiI6MTcxNjExMDA1NSwicGF0aCI6Ii82NDcyNDE1Ny8zMzE4NTk4OTEtMDI3NDg0ZDQtMGQ1NC00YTViLWJkZGMtMzNmZjQ0ZThjODA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDA5MTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2Yjg1ZjQzMGY1ODFjNGJlNDA4MTY4MTk5NjZjNWJkZDBjNDYzNTgyY2MxZTI1NDhkZWJkOGI4MGIzYjBiMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aCynnz5kLBW60DP4l6ylipanYxDoiNOqK_RxExpTZ6A)
> 
> * mobile version

Can we make it 1 column on mobile?

bcho892 commented:
Looks pretty good - not the easiest ticket so gj!

Will boot it up locally soon to verify

zlrkw11 commented:
> > ![image](https://private-user-images.githubusercontent.com/64724157/331859891-027484d4-0d54-4a5b-bddc-33ff44e8c804.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxMTAzNTUsIm5iZiI6MTcxNjExMDA1NSwicGF0aCI6Ii82NDcyNDE1Ny8zMzE4NTk4OTEtMDI3NDg0ZDQtMGQ1NC00YTViLWJkZGMtMzNmZjQ0ZThjODA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDA5MTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2Yjg1ZjQzMGY1ODFjNGJlNDA4MTY4MTk5NjZjNWJkZDBjNDYzNTgyY2MxZTI1NDhkZWJkOGI4MGIzYjBiMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aCynnz5kLBW60DP4l6ylipanYxDoiNOqK_RxExpTZ6A)
> > 
> > * mobile version
> 
> Can we make it 1 column on mobile?

![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/0bbe3dea-3996-479d-b940-28396ddeece5)

- changes applied.

bcho892 commented on client/src/pages/Profile/Profile.tsx:
```diff
@@ -0,0 +1,133 @@
+                  />
+                  <Field
+                    subtitle="Date of birth"
+                    description={`${currentUserData?.date_of_birth}`}
```
you are getting `[object Object]` because you aren't formatting the date correctly. `date_of_birth` contains a `seconds` field, read up on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date to see how you can make a date string

zlrkw11 replied:
done

bcho892 commented on client/src/pages/Profile/Profile.tsx:
```diff
@@ -0,0 +1,140 @@
+  const [{ currentUser }] = useAppData()
+
+  function toDateTime(secs?: number) {
+    const t = new Date(1970, 0, 1) // Epoch
```
is this right?????

zlrkw11 replied:
I thought the default date is 1970, 01, 01. And you add seconds to it. Will change it

zlrkw11 commented:
![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/df3b9dbd-89aa-46f6-b9ce-eef6cfa2121a)
- made sign out button visible while in sm (mobile viewport)
- display date correctly

bcho892 reviewed (changes requested):
![image](https://github.com/UoaWDCC/uasc-web/assets/100653148/af5b1f69-3992-4ef0-9549-fcc5eac6db43)

This seems very squished, you may want to set an appropriate max width and/or add appropriate paddings/margins to the parent container

bcho892 commented:
![image](https://github.com/UoaWDCC/uasc-web/assets/100653148/63a01d4b-00a7-4f08-b444-afa1ed7a8fa2)

Also it does not scroll and gets cut off on mobile view

bcho892 commented on client/src/pages/Profile/Profile.tsx:
```diff
@@ -0,0 +1,140 @@
+          <div className="grid w-full gap-4">
+            <ProfileInformationPanel title="Personal details" onEdit={() => {}}>
+              <div className="flex w-full flex-col gap-4">
+                <div className="grid grid-cols-4">
```
you may want to reduce the cols on smaller screens

zlrkw11 commented:
![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/e8a54d39-eddf-465b-a873-d2f1e69bf7b1)
- fixing up the footer scrolling issue. Related to button's margin left value, now always pushing to the end, more responsive.
![image](https://github.com/UoaWDCC/uasc-web/assets/64724157/270b1bb9-3cbe-4a4d-b58c-fbe76152a8c4)
- fixed the problem with not being able to scroll vertically, by making a new responsive background image component based on Jefferey's full background image component. 
- fixed the date's display
